client/src/Pos/Client/KeyStorage.hs

```diff
@@ -45,7 +46,9 @@ class Monad m => MonadKeysRead m where
     getSecret :: m UserSecret
 
 class MonadKeysRead m => MonadKeys m where
+    modifySecretNew :: State UserSecret a -> m a
```

Comment:
That's probably not the most beautiful thing one could do, but I wanted to make mimimal amount of changes.